client/src/partials/finance/cashflow/config.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+angular.module('bhima.controllers')
+.controller('CashflowConfigController', CashflowConfigController);
+
+// dependencies injection
+CashflowConfigController.$inject = [ '$state', 'CashboxService', 'NotifyService' ];
+
+/**
+ * Cashflow config controller
+ *
+ * @description
+ * This controller is responsible of cash flow report, that report include
+ * all incomes minus all depenses
+ */
+function CashflowConfigController($state, Cashbox, Notify) {
+  var vm = this;
+
+  // expose to the view
+  vm.generate = generate;
+
+  /** generate cashflow report */
+  function generate() {
+    if (!vm.cashbox || !vm.dateFrom || !vm.dateTo) { return ; }
+    var params = {
+      dateFrom : vm.dateFrom,
+      dateTo   : vm.dateTo,
+      cashbox  : vm.cashbox
+    };
+    $state.go('cashflow.report', params);
+  }
+
+  /** init */
+  (function init() {
```

Comment:
In controllers, you do not need an [IIFE](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression) - there is no need to namespace things.  It just make the code harder to read.

If you want to clearly have an `init()` function, you can use angular's `$onInit()` inside components.  It doesn't work for non-component controllers (as far as I know).  See [this documentation](https://docs.angularjs.org/guide/component#component-based-application-architecture).